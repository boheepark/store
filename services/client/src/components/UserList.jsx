import React from 'react';
import {Table} from 'react-bootstrap';

const UserList = (props) => {
  return (
    <div>
      <h1>All Users</h1>
      <hr/>
      <br/>
      <Table striped bordered condensed hover>
        <thead>
        <tr>
          <th>User ID</th>
          <th>Username</th>
          <th>Email</th>
          <th>Active</th>
          <th>Admin</th>
          <th>Created Date</th>
        </tr>
        </thead>
        <tbody>
          {
            props.users.map((user) => {
              return (
                <tr key={user.id}>
                  <td>{user.id}</td>
                  <td>{user.username}</td>
                  <td>{user.email}</td>
                  <td>{String(user.active)}</td>
                  <td>{String(user.admin)}</td>
                  <td>{user.created_at}</td>
                </tr>
              );
            })
          }
        </tbody>
      </Table>
    </div>
  );
};

export default UserList;
