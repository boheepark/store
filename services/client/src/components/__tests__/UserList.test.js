import React from 'react';
import {shallow} from "enzyme";
import renderer from 'react-test-renderer';

import UserList from "../UserList";


const date = new Date(1482363367071);
const users = [
  {
    'id': 1,
    'username': 'test',
    'email': 'test@email.com',
    'active': true,
    'admin': false,
    'created_at': date.toUTCString()
  },
  {
    'id': 2,
    'username': 'test2',
    'email': 'test2@email.com',
    'active': true,
    'admin': false,
    'created_at': date.toUTCString()
  }
];


test('UserList renders properly', () => {
  const wrapper = shallow(<UserList users={users}/>);

  const h1 = wrapper.find('h1');
  expect(h1.get(0).props.children).toBe('All Users');

  const table = wrapper.find('Table');
  expect(table.length).toBe(1);
  expect(table.get(0).props.striped).toBe(true);
  expect(table.get(0).props.bordered).toBe(true);
  expect(table.get(0).props.condensed).toBe(true);
  expect(table.get(0).props.hover).toBe(true);

  const thead = wrapper.find('thead');
  expect(thead.length).toBe(1);

  const th = wrapper.find('th');
  expect(th.length).toBe(6);
  expect(th.get(0).props.children).toBe('User ID');
  expect(th.get(1).props.children).toBe('Username');
  expect(th.get(2).props.children).toBe('Email');
  expect(th.get(3).props.children).toBe('Active');
  expect(th.get(4).props.children).toBe('Admin');
  expect(th.get(5).props.children).toBe('Created Date');

  const tbody = wrapper.find('tbody');
  expect(tbody.length).toBe(1);
  expect(wrapper.find('tbody > tr').length).toBe(2);

  const td = wrapper.find('tbody > tr > td');
  expect(td.length).toBe(12);
  expect(td.get(0).props.children).toBe(1);
  expect(td.get(1).props.children).toBe('test');
  expect(td.get(2).props.children).toBe('test@email.com');
  expect(td.get(3).props.children).toBe('true');
  expect(td.get(4).props.children).toBe('false');
  expect(td.get(5).props.children).toBe(date.toUTCString());
});


test('UserList renders a snapshot properly', () => {
  const tree = renderer.create(<UserList users={users}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
