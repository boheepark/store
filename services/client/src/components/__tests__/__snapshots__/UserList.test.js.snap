// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList renders a snapshot properly 1`] = `
<div>
  <h1>
    All Users
  </h1>
  <hr />
  <br />
  <table
    className="table table-striped table-bordered table-condensed table-hover"
  >
    <thead>
      <tr>
        <th>
          User ID
        </th>
        <th>
          Username
        </th>
        <th>
          Email
        </th>
        <th>
          Active
        </th>
        <th>
          Admin
        </th>
        <th>
          Created Date
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          1
        </td>
        <td>
          test
        </td>
        <td>
          test@email.com
        </td>
        <td>
          true
        </td>
        <td>
          false
        </td>
        <td>
          Wed, 21 Dec 2016 23:36:07 GMT
        </td>
      </tr>
      <tr>
        <td>
          2
        </td>
        <td>
          test2
        </td>
        <td>
          test2@email.com
        </td>
        <td>
          true
        </td>
        <td>
          false
        </td>
        <td>
          Wed, 21 Dec 2016 23:36:07 GMT
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
